### PR TITLE
 glpi-agent: init at 1.11

### LINKED ISSUE #361741: Package request: glpi-agent
## Metadata

* homepage URL: https://glpi-project.org/
* source URL: https://github.com/glpi-project/glpi-agent
* license: gpl2+
* platforms: windows, linux, darwin

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by liberodark**
Fix : https://github.com/NixOS/nixpkgs/issues/361741

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by liberodark** [comment]
Also this need a backport to 24.11 if is possible is tested and used on 24.11

File: pkgs/by-name/gl/glpi-agent/package.nix [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}:
+
+perlPackages.buildPerlPackage rec {
+  pname = "GLPI-Agent";
+  version = "1.11";
+
+  src = fetchFromGitHub {
+    owner = "glpi-project";
+    repo = "glpi-agent";
+    rev = version;
```
Comment (GaetanLepage) [inline pkgs/by-name/gl/glpi-agent/package.nix:21]:
```suggestion
    tag = version;
```

File: pkgs/by-name/gl/glpi-agent/package.nix [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      fi
+    done
+  '';
+
+  outputs = [ "out" ];
+
+  meta = {
+    homepage = "https://glpi-project.org/";
+    description = "GLPI unified Agent for UNIX, Linux, Windows and MacOSX";
+    license = lib.licenses.gpl2Plus;
```
Comment (GaetanLepage) [inline pkgs/by-name/gl/glpi-agent/package.nix:120]:
```suggestion
    description = "GLPI unified Agent for UNIX, Linux, Windows and MacOSX";
    changelog = "https://github.com/glpi-project/glpi-agent/releases/tag/${version}";
    license = lib.licenses.gpl2Plus;
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Minor comments. I am not too familiar with the perl ecosystem.

File: pkgs/by-name/gl/glpi-agent/package.nix [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    repo = "glpi-agent";
+    tag = version;
+    hash = "sha256-WdQ+/ZnMCRqLZK64oJNoR9dtMPq+CghsA8NUwt3EpjA=";
+  };
+
+  postPatch = ''
+    patchShebangs bin
+
+    substituteInPlace "lib/GLPI/Agent/Tools/Linux.pm" \
+      --replace /sbin/ip ${iproute2}/sbin/ip
```
Comment (SuperSandro2000) [inline pkgs/by-name/gl/glpi-agent/package.nix:29]:
```suggestion
      --replace-fail /sbin/ip ${iproute2}/sbin/ip
```

File: pkgs/by-name/gl/glpi-agent/package.nix [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    substituteInPlace "lib/GLPI/Agent/Task/Inventory/Linux/Networks.pm" \
+      --replace /sbin/ip ${iproute2}/sbin/ip
```
Comment (SuperSandro2000) [inline pkgs/by-name/gl/glpi-agent/package.nix:31]:
```suggestion
      --replace-fail /sbin/ip ${iproute2}/sbin/ip
```

File: pkgs/by-name/gl/glpi-agent/package.nix [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  '';
+
+  buildTools = [ ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/gl/glpi-agent/package.nix:34]:
```suggestion
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/by-name/gl/glpi-agent/package.nix [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            nettools
+            iproute2
+          ]
+        }
+      fi
+    done
+  '';
+
+  outputs = [ "out" ];
+
```
[the remark this answers is not in the document]
Comment (liberodark) [reply pkgs/by-name/gl/glpi-agent/package.nix]: Yes

File: pkgs/by-name/gl/glpi-agent/package.nix [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    done
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+
+  versionCheckProgramArg = [ "--version" ];
+
+  doInstallCheck = true;
```
[the remark this answers is not in the document]
Comment (liberodark) [reply pkgs/by-name/gl/glpi-agent/package.nix:124]: I think is not necessary.

File: pkgs/by-name/gl/glpi-agent/package.nix [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    description = "GLPI unified Agent for UNIX, Linux, Windows and MacOSX";
+    homepage = "https://glpi-project.org/";
+    changelog = "https://github.com/glpi-project/glpi-agent/releases/tag/${src.tag}";
+    license = lib.licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [ liberodark ];
+    badPlatforms = [
+      # I don't have mac to try this build.
+      lib.systems.inspect.patterns.isDarwin
+    ];
+    mainProgram = "glpi-agent";
```
[the remark this answers is not in the document]
Comment (liberodark) [reply pkgs/by-name/gl/glpi-agent/package.nix:140]: Why ?

File: pkgs/by-name/gl/glpi-agent/package.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (liberodark) [reply pkgs/by-name/gl/glpi-agent/package.nix:124]: Have fix issue on your can build on your side.

**Review by GaetanLepage — APPROVED** [review]

**Pull request merged**